mailpoet/tests/integration/Newsletter/Scheduler/PostNotificationTest.php

```diff
@@ -289,17 +304,25 @@ public function testSchedulerWontRunIfUnsentNotificationHistoryExists() {
     ]);
 
     $notificationHistory = new NewsletterEntity();
-    $notificationHistory->setType(Newsletter::TYPE_NOTIFICATION_HISTORY);
-    $notificationHistory->setStatus(Newsletter::STATUS_SENDING);
+    $notificationHistory->setType(NewsletterEntity::TYPE_NOTIFICATION_HISTORY);
+    $notificationHistory->setStatus(NewsletterEntity::STATUS_SENDING);
     $notificationHistory->setParent($newsletter);
     $notificationHistory->setSubject($newsletter->getSubject());
     $this->newslettersRepository->persist($notificationHistory);
     $this->newslettersRepository->flush();
 
-    $sendingTask = SendingTask::create();
-    $sendingTask->newsletterId = $notificationHistory->getId();
-    $sendingTask->status = SendingQueue::STATUS_SCHEDULED;
-    $sendingTask->save();
+    $task = new ScheduledTaskEntity();
+    $task->setType(ScheduledTaskEntity::DEFAULT_TASK_TYPE);
+    $task->setStatus( SendingQueueEntity::STATUS_SCHEDULED);
+    $this->scheduledTasksRepository->persist($task);
+    $this->scheduledTasksRepository->flush();
+
+    $this->sendingQueuesRepository->flush();
+    $sendingQueue = new SendingQueueEntity();
```

Comment:
I have created a SendingQueue factory in [another PR](https://github.com/mailpoet/mailpoet/pull/4305/commits/a0cf7a817de934f720a3c907f0c7848fd62dbb1a). Unfortunately, it is not merged yet.